pkg/generate/app/pipeline.go

```diff
@@ -311,6 +311,30 @@ func objectMetaData(raw interface{}) (runtime.Object, *kapi.ObjectMeta, error) {
 	return obj, meta, nil
 }
 
+type acceptBuildConfigs struct {
+	typer runtime.ObjectTyper
+}
+
+// Accept accepts just BuildConfigs
+func (a *acceptBuildConfigs) Accept(from interface{}) bool {
+	obj, _, err := objectMetaData(from)
+	if err != nil {
+		return false
+	}
+	_, kind, err := a.typer.ObjectVersionAndKind(obj)
+	if err != nil {
+		return false
+	}
+	return kind == "BuildConfig"
```

Comment:
It's probably not going to be the default - a user should have to give us a --to-docker flag (or similar).  Ideally I'd like to reserve "--to" to be an Openshift image tag (so you can easily spin up a new build pointing to that tag, vs using a default name).

> On Jun 3, 2015, at 6:04 PM, Maciej Szulik <notifications@github.com> wrote:
> 
> In pkg/generate/app/pipeline.go:
> 
> > @@ -311,6 +311,30 @@ func objectMetaData(raw interface{}) (runtime.Object, *kapi.ObjectMeta, error) {
> >  	return obj, meta, nil
> >  }
> >  
> > +type acceptBuildConfigs struct {
> > +	typer runtime.ObjectTyper
> > +}
> > +
> > +// Accept accepts just BuildConfigs
> > +func (a *acceptBuildConfigs) Accept(from interface{}) bool {
> > +	obj, _, err := objectMetaData(from)
> > +	if err != nil {
> > +		return false
> > +	}
> > +	_, kind, err := a.typer.ObjectVersionAndKind(obj)
> > +	if err != nil {
> > +		return false
> > +	}
> > +	return kind == "BuildConfig"
> We do support pushes to external repo -> DockerImage.
> 
> —
> Reply to this email directly or view it on GitHub.
>